### PR TITLE
fix: remove lock file on process exit and checkpoint WAL on close

### PR DESCRIPTION
**Pull request by lsm**
## Summary

- **Lock file not cleaned up on SIGTERM/SIGHUP**: `DatabaseLock.acquire()` now registers a synchronous `process.on('exit')` fallback handler that removes the lock file even when async signal handlers don't fire (e.g. Bun kills the process on Ctrl+C during `bun run`). `release()` deregisters the handler via `process.removeListener()` so it doesn't double-fire after graceful shutdown.
- **WAL checkpoint on close**: `DatabaseCore.close()` now runs `PRAGMA wal_checkpoint(TRUNCATE)` before `db.close()` to flush all write-ahead log frames to the main DB file, preventing data loss on abrupt termination.

## Tests

- New `database-lock.test.ts` (11 tests): acquire creates lock + exit listener; release removes both; idempotency; in-memory bypass; exit handler directly removes lock file.
- Extended `database-core.test.ts`: WAL file is empty/absent after `close()`; lock file released on `close()`.
- Extended `daemon-app-cleanup.test.ts`: lock file present during run, absent after `cleanup()`; exit listener count decreases after cleanup.

### CODE REVIEW

File: packages/daemon/tests/unit/1-core/core/daemon-app-cleanup.test.ts [hunk 1]
```diff
@@ -299,4 +302,77 @@ describe('Daemon App Cleanup', () => {
...
+			tmpDbDir = join(
+				tmpdir(),
+				`neokai-cleanup-test-${Date.now()}-${Math.random().toString(36).slice(2)}`
+			);
+			mkdirSync(tmpDbDir, { recursive: true });
+		});
+
+		afterEach(() => {
+			try {
+				const { rmSync } = require('node:fs');
```
Comment (lsm) [inline packages/daemon/tests/unit/1-core/core/daemon-app-cleanup.test.ts]: Minor style nitpick: this uses `require('node:fs')` while the rest of the file uses top-level ES module imports (line 12: `import { existsSync, unlinkSync, mkdirSync } from 'node:fs'`). Consider adding `rmSync` to the existing import at the top of the file for consistency.

**Review by lsm — COMMENTED** [review]
## Review: Fix database lock not removed on stop and add graceful shutdown WAL checkpoint

**Verdict: Approve** ✅

Clean, focused fix for two real shutdown bugs. The implementation is correct and well-tested.

### What's good

- **`process.on('exit')` fallback** (`database-lock.ts`): Right approach. Synchronous exit handlers always fire, unlike async signal handlers that can be skipped by Bun or SIGTERM. The closure captures `lockPath` (not `this`) so it works even if the instance is partially GC'd. Deregistration in `release()` via `process.removeListener()` prevents double-unlink after graceful shutdown.
- **Idempotent `acquire()`**: Doesn't register duplicate listeners on repeated calls — verified by test.
- **WAL checkpoint in `close()`** (`database-core.ts`): `PRAGMA wal_checkpoint(TRUNCATE)` before `db.close()` is the right call. TRUNCATE mode resets the WAL file, and the try/catch handles the edge case where the DB is already in an error state.
- **Test coverage**: 15 new tests across 3 files covering acquire/release/exit-handler behavior, WAL checkpoint verification, and end-to-end lock file lifecycle through daemon startup/cleanup.
- **No overreach**: Stale-lock detection, signal handlers in main.ts/dev-server.ts, and `isProcessAlive` are all untouched as instructed.

### Minor nit (non-blocking)

- `require('node:fs')` in the afterEach of `daemon-app-cleanup.test.ts` — inconsistent with the file's ES module imports at the top. (Comment posted on line 319.)

### Pre-existing note (not introduced by this PR)

- `DatabaseCore.close()` has no null guard on `this.db` — if called before `initialize()`, `this.db.close()` would throw. This is pre-existing (the same call was there before) and the normal lifecycle makes it safe, but a defensive `if (this.db === null) return` would be a small improvement for robustness.

File: packages/daemon/tests/unit/1-core/core/daemon-app-cleanup.test.ts (hunk 1 above)
Comment (lsm) [reply packages/daemon/tests/unit/1-core/core/daemon-app-cleanup.test.ts]: Fixed — added `rmSync` to the existing top-level import on line 12 and removed the inline `require()`. Clean now.
